mlir/utils/jenkins/miopen-tests/miopen_validate.sh

```diff
@@ -160,12 +173,35 @@ function run_tests() {
     fi
 }
 
-function main() {
-    parse_options "$@"
+function run_tests_for_a_config() {
     construct_fixed_args
     create_configs
     setup_environment
     run_tests
 }
 
+function tune_all_configs() {
+    for t in ${ALL_DTYPES[@]}; do
+        for d in ${ALL_DIRECTIONS[@]}; do
+            for l in ${ALL_LAYOUTS[@]}; do
+                DTYPE=$t
+                DIRECTION=$d
+                LAYOUT=$l
+                run_tests_for_a_config
+            done
+        done
+    done
+}
+
+function main() {
+    clean_miopen_caches
+    parse_options "$@"
+    get_configs
+    if [[ $TUNEALL == 1 ]]; then
```

Comment:
The tuneall machinery is so you can have the Jenkinsfile run all the tuning on one node?

Reply:
Yes. I want the tuning to generate one user perfDb which I can use for performance report. If the tuning is run in parallel, there will be multiple perfDbs at different nodes, making them hard to be used.